core/extractors/extractors_latex.py

```
@@ -0,0 +1,494 @@
+            return self._empty_result(latex_path, str(e))
+    
+    def _extract_from_tar_gz(self, latex_path: Path, start_time: datetime) -> Dict[str, Any]:
+        """Extract from tar.gz archive (multiple files)."""
+        with tempfile.TemporaryDirectory() as temp_dir:
+            temp_path = Path(temp_dir)
+            
+            try:
+                # Extract the tar.gz file
+                with gzip.open(latex_path, 'rb') as gz:
+                    with tarfile.open(fileobj=gz, mode='r') as tar:
+                        tar.extractall(temp_path)
```

Ruff (0.12.2)

Uses of `tarfile.extractall()`

(S202)